assets/js/components/surveys/CurrentSurvey.test.js

```diff
@@ -0,0 +1,412 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import CurrentSurvey from './CurrentSurvey';
+import { render, fireEvent, createTestRegistry } from '../../../../tests/js/test-utils';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { CORE_FORMS } from '../../googlesitekit/datastore/forms/constants';
+import * as fixtures from './__fixtures__';
+
+describe( 'CurrentSurvey', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	it( 'should render a survey when one exists in the datastore', async () => {
+		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );
+
+		fetchMock.postOnce( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );
+
+		const { container } = render( <CurrentSurvey />, { registry } );
+
+		expect( fetchMock ).toHaveFetched( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/ );
+
+		expect( container ).toMatchSnapshot();
+	} );
+
+	it( "should render a rating question when the `question_type` is 'rating'", async () => {
+		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );
+
+		fetchMock.postOnce( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );
+
+		const { container } = render( <CurrentSurvey />, { registry } );
+
+		expect( fetchMock ).toHaveFetched( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/ );
+
+		expect( container ).toMatchSnapshot();
+	} );
+
+	it( 'should render nothing when the `question_type` is unknown', async () => {
+		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.invalidQuestionTypeSurvey, { triggerID: 'jestSurvey' } );
+
+		fetchMock.postOnce( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );
+
+		const { container } = render( <CurrentSurvey />, { registry } );
+
+		expect( container ).toBeEmptyDOMElement();
+	} );
+
+	it( "should send a 'survey_shown' event on mount", async () => {
+		registry.dispatch( CORE_USER ).receiveTriggerSurvey( fixtures.singleQuestionSurvey, { triggerID: 'jestSurvey' } );
+
+		fetchMock.postOnce( /^\/google-site-kit\/v1\/core\/user\/data\/survey-event/, { body: {}, status: 200 } );
+
+		const { rerender } = render( <CurrentSurvey />, { registry } );
+
+		expect( fetchMock ).toHaveFetched(
+			'/google-site-kit/v1/core/user/data/survey-event?_locale=user',
+			{
+				body: {
+					data: {
+						event: { survey_shown: {} },
+						session: fixtures.singleQuestionSurvey.session,
+					},
+				},
+				credentials: 'include',
+				headers: {
+					Accept: 'application/json, */*;q=0.1',
+					'Content-Type': 'application/json',
+				},
+				method: 'POST',
+			}
+		);
+
+		fetchMock.reset();
```

Comment:
This is probably not what we want to do right here, as [`reset`](http://www.wheresrhys.co.uk/fetch-mock/#api-lifecyclerestore_reset) "Resets fetch() to its unstubbed state".

I think what we want instead is `resetHistory()`

```suggestion
		fetchMock.resetHistory();
```